### ISSUE TITLE
Hardware Tessellation was grayed-out in OpenGL backend

### ISSUE THREAD

**Issue by WarmestKrafter** [hrydgard/ppsspp#18496]
Why does it got grayed-out in OpenGL? I want to run spline/bezier curves on the GPU instead of the CPU so it won't affect my performance.
![Screenshot_2023-12-09-18-23-49-859_org ppsspp ppsspp](https://github.com/hrydgard/ppsspp/assets/112315947/32a5aed3-cd97-4284-8eb9-8f4ec99d78b5)

**Comment by hrydgard** [comment]
We require certain OpenGL features to be available to make that selectable (when it's disabled, even if it looks check, it of course won't be used).

Presumably your GPU doesn't support said features. Here's the check code:

```c
int maxVertexTextureImageUnits = gl_extensions.maxVertexTextureUnits;
bool vertexTexture = maxVertexTextureImageUnits >= 3; // At least 3 for hardware tessellation

bool textureFloat = gl_extensions.ARB_texture_float || gl_extensions.OES_texture_float;
bool hasTexelFetch = gl_extensions.GLES3 || (!gl_extensions.IsGLES && gl_extensions.VersionGEThan(3, 3, 0)) || gl_extensions.EXT_gpu_shader4;

return vertexTexture && textureFloat && hasTexelFetch;
```

So, we require support for float textures, and texturing from the vertex shader, and OpenGL ES 3.0.

Can you show the GPU information on "Settings / Tools / System Information" on your device?

**Comment by LunaMoo** [comment]
Just to note, in practice curves done on GPU being faster than on CPU is situational depending on your hardware actual bottlenecks and not always faster, in many situations they aren't, but what they always are is the fact that they're less accurate and still known to cause problems with some games like for example Test Drive Unlimited. On top of that, just a minor fraction of PSP games are actually using those and there's maybe like 2-3 games which makes use of it enough to even matter for performance and commonly those are ones where hardware tesselation causes glitches.

 In other words, the setting is overrated by many users purely based on "tesselation" in the name and should not be something to be sad about missing out on older hardware.

**Comment by anr2me** [comment]
We probably need to set it to unticked if it's not going to be used/not supported, because common people probably think it's always enabled and can't be disabled when seeing a ticked checkbox that can't be changed :)

**Comment by WarmestKrafter** [comment]
> We probably need to set it to unticked if it's not going to be used/not supported, because common people probably think it's always enabled and can't be disabled when seeing a ticked checkbox that can't be changed :)

I posted the wrong image, sorry.

**Comment by hrydgard** [comment]
Either way there's not a lot to do here, and you still haven't posted the picture I requested from `Settings / Tools / System Information`. Closing.

**Comment by Crisanto47** [comment]
Closed this https://github.com/hrydgard/ppsspp/pull/20369/commits/4e25eadd23618bc3278164b4948566f2163d689d
